go/vt/vtgate/planbuilder/operators/delete.go

```diff
@@ -88,32 +88,33 @@ func (d *Delete) ShortDescription() string {
 	return fmt.Sprintf("%s.%s%s%s", d.Target.VTable.Keyspace.Name, d.Target.VTable.Name.String(), orderBy, limit)
 }
 
-func createOperatorFromDelete(ctx *plancontext.PlanningContext, deleteStmt *sqlparser.Delete) Operator {
+func createOperatorFromDelete(ctx *plancontext.PlanningContext, deleteStmt *sqlparser.Delete) (op Operator) {
 	delClone := sqlparser.CloneRefOfDelete(deleteStmt)
 
 	delOp := createDeleteOperator(ctx, deleteStmt)
+	op = delOp
 
 	if deleteStmt.Comments != nil {
-		delOp = &LockAndComment{
-			Source:   delOp,
+		op = &LockAndComment{
+			Source:   op,
 			Comments: deleteStmt.Comments,
 		}
 	}
 
 	childFks := ctx.SemTable.GetChildForeignKeysList()
 	// If there are no foreign key constraints, then we don't need to do anything.
 	if len(childFks) == 0 {
-		return delOp
+		return
```

Comment:
Nit, but I prefer to still return with an actual named argument here to be explicit (so `op`).